### PR TITLE
Fixed money text alignment.

### PR DESCRIPTION
**Pull request by scirelli**
# Description
* Attempted to fix Issue #136
* Updated based on comments from last PR #141

Added a new method

# Testing
* See issue #136 to reproduce the draw over.
* Edit your prevShip.ini, change the money value to something over 9999. It should no longer render over the icon, and remain left aligned.
* Change it to the max value 2147483647, it should still not draw over the icon.

# Notes
I ran the default Intellij's code formatter. I have my tabs set to 4 spaces. You don't have any documentation on code format. I see 2 space tabs, and I think I see 4 space ones as well. let me know which one you prefer. 

### Pre Pull Request Checklist:

- [x] Code has been scanned with [SonarLint](http://www.sonarlint.org/intellij/)
- [ ] Methods have been annotated with @ Nullable and @ Nonnull ([Read more](https://github.com/MovingBlocks/DestinationSol/wiki/@Nullable-&-@Nonnull-Quickstart))
- [x] There are no errors present in the project
- [x] Code has been formatted and indented
- [ ] Methods have appropriate Javadoc ([How to write Javadoc](http://www.oracle.com/technetwork/java/javase/documentation/index-137868.html))

### CODE REVIEW

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: main/src/org/destinationsol/CommonDrawer.java [hunk 1]
```diff
@@ -27,126 +27,142 @@
 import com.badlogic.gdx.math.Vector2;
 import org.destinationsol.common.SolMath;
 import org.destinationsol.files.FileManager;
+import org.destinationsol.TextAlignment;
```
Comment (scirelli) [inline main/src/org/destinationsol/CommonDrawer.java]: Not sure if you wanted me to create one or use the javafx one.

File: main/src/org/destinationsol/CommonDrawer.java [hunk 2]
```diff
@@ -27,126 +27,142 @@
...
+    public void end() {
+        myTextureChecker.onEnd();
+        mySpriteBatch.end();
+    }
+
+    public void drawString(String s, float x, float y, float fontSize, boolean centered, Color col) {
+        drawString(s, x, y, fontSize, TextAlignment.CENTER, centered, col);
+    }
+
+    public void drawString(String s, float x, float y, float fontSize, TextAlignment align, boolean verticalCentering, Color col) {
```
Comment (scirelli) [inline main/src/org/destinationsol/CommonDrawer.java]: Changes are here

File: main/src/org/destinationsol/game/screens/MainScreen.java [hunk 3]
```diff
@@ -40,493 +39,501 @@
...
-    public String text;
-    public Vector2 pos = new Vector2();
-    public final Color color;
+        public void draw(UiDrawer uiDrawer) {
+            uiDrawer.drawString(text, pos.x, pos.y, FontSize.HUD, true, color);
+        }
 
-    public TextPlace(Color col) {
-      color = new Color(col);
+        public void draw(UiDrawer uiDrawer, TextAlignment align) {
```
Comment (scirelli) [inline main/src/org/destinationsol/game/screens/MainScreen.java]: Added this method.

**Comment by Cervator** [comment]
Adding @vampcat as a reviewer for a test - likely will be helping out some from the admin side of things :-)

File: main/src/org/destinationsol/CommonDrawer.java (hunk 1 above)
Comment (vampcat) [inline main/src/org/destinationsol/CommonDrawer.java]: Either would have worked, but I'll glad you decided to go with this.

File: main/src/org/destinationsol/TextAlignment.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2016 MovingBlocks
```
Comment (vampcat) [inline main/src/org/destinationsol/TextAlignment.java]: It's 2017, my good sir :)

Just a minor nitpick, but bump up the year whenever you work on a file.

File: main/src/org/destinationsol/TextAlignment.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.destinationsol;
+
+/**
+ * Created by scirelli on 4/1/17.
+ */
```
Comment (vampcat) [inline main/src/org/destinationsol/TextAlignment.java]: As a general rule, removing the auto-generated author tag is recommended.

**Comment by GooeyHub** [comment]
Uh oh, something went wrong with the build. Need to check on that

**Comment by scirelli** [comment]
@vampcat Done.

**Comment by vampcat** [comment]
@GooeyHub: test this please

**Comment by Cervator** [comment]
@GooeyHub: test this please

(I think there's a user list in Jenkins we need to tweak)

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by vampcat** [comment]
@scirelli There we go.
Hopefully we can streamline the PR process a lot more in the coming days :D

**Comment by Cervator** [comment]
Belated update: I found a small alignment issue this introduced on the interaction screens with a station (buy, sell, mercenaries etc). Had to add the left alignment thing to a few spots. Did so in https://github.com/MovingBlocks/DestinationSol/pull/143/commits/8b98a51170391573645283aa6502a79d8aaf3435 although I forgot to note it in the commit message or split it out into its own commit.

There maaaay be other spots I missed but hopefully that'll take care of it :-)

**Pull request merged**